installer/src/systemd_boot/mod.rs

```diff
@@ -165,9 +165,18 @@ fn remove_old_files(generations: &[Generation], path: &Path) -> Result<()> {
         let f = entry?.path();
         let name = f.file_name().ok_or("filename terminated in ..")?;
 
+        // fwupd puts its own files under "fw" directory, and doesn't remove them
+        // This avoids unhelpful error messages from fs::remove_file later
+        if name == "fw" && f.is_dir() {
+            trace!("Skipping firmware update directory \"fw\"");
+            continue;
+        }
+
         if !required_filenames.iter().any(|e| e == name) {
             trace!("removing kernel/initrd file {:?}", f);
-            fs::remove_file(f)?;
+            if let Err(e) = fs::remove_file(f.clone()) {
```

Comment:
I think this should continue to be a hard error. IMHO, I don't think it's a good idea to allow potentially unsigned files if they fail to be removed.

Reply:
So maybe adding `std::process::exit(1)` after `eprintln!()`? The error message from `fs::remove_file()?` is really unhelpful.